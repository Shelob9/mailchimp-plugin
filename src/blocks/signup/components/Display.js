import React from 'react';
import {mailChimpTestForm} from "../../../components/mailChimpTestForm.fixture";
import MailChimpForm from "../../../components/MailChimpForm";

export const Display = ({listId,form,Fallback}) => {
	if( listId ){
		return <MailChimpForm
			form={form}
			onSubmit={(values) => alert(JSON.stringify(values))}
		/>
	}

	return <Fallback/>
};

