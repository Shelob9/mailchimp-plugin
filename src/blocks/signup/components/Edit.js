import React from 'react';
import {HorizontalForm} from '@calderajs/forms';

export const Edit = ({onChangeListId, listId, listFields, instanceId}) => {
	const initialValues = {};
	function getListIdFromValues(values){
		return '';
	}
	return (
		<div>
			<HorizontalForm
				fields={listFields}
				initialValues={initialValues}
				onClose={() => alert('close')}
				onChange={(values) => {
					console.log(values);
					onChangeListId(getListIdFromValues(values))
				}}
				instanceId={instanceId}
			/>
		</div>
	);
}



