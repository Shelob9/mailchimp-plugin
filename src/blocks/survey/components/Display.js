import {createElement} from '@wordpress/element';
import CalderaMailChimpSurveyForm from "../../../components/CalderaMailChimpSurveyForm";



export const Display = ({listId, apiRoot,token,Fallback}) => {
	if( listId && '---' !== listId ){
		return <CalderaMailChimpSurveyForm
			apiRoot={apiRoot}
			token={token}
			listId={listId}
			onSubmit={(values) => alert(JSON.stringify(values))}
		/>

	}
	return <Fallback/>
};


