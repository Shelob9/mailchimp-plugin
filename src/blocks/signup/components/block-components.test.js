import renderer from 'react-test-renderer';
import {Display} from "./Display";
import {Edit} from "./Edit";
import {mailChimpTestForm} from "../../../components/mailChimpTestForm.fixture";

describe('Tests working', () => {
	it('should work', () => {
		expect(1).toBe(1);
	});
});

describe('Display component', () => {
	it( 'Shows mailchimp form', () => {
		const component = renderer.create(
			<Display Fallback={<div>Fallback</div>} form={{mailChimpTestForm} } listId={'listId'}/>
		);
		expect( component.toJSON()).toMatchSnapshot();
	});

	it( 'shows fallback if no list id', () => {
		const Fallback  = () => (<div>Fallback</div>);
		const component = renderer.create(
			<Display Fallback={Fallback} form={{mailChimpTestForm} } />
		);
		expect( component.toJSON()).toMatchSnapshot();
	});

	const listFieldUiConfig= [];
	it( 'shows edit form', () => {
		const onChange = jest.fn();
		const component = renderer.create(
			<Edit
				listId={'listId'}
				onChangeListId={onChange}
				listFields={listFieldUiConfig}
			/>
		);
		expect( component.toJSON()).toMatchSnapshot();
	});
});



