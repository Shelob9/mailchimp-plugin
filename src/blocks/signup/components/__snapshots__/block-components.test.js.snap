// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component Shows mailchimp form 1`] = `
<div>
  <div
    className="css-16tm1g7-wrapper-wrapper"
  >
    <div
      className="css-1anlcob-pacmanStyle"
    />
    <div
      className="css-1oj2eaa-pacmanStyle"
    />
    <div
      className="css-1y2ttxx-ballStyle"
    />
    <div
      className="css-im5qma-ballStyle"
    />
    <div
      className="css-10tjdos-ballStyle"
    />
    <div
      className="css-x3nw5t-ballStyle"
    />
  </div>
</div>
`;

exports[`Display component shows edit form 1`] = `
<div>
  <div
    className="caldera"
  >
    <form
      className="horizontal-form-undefined caldera-form"
      onSubmit={[Function]}
    />
  </div>
</div>
`;

exports[`Display component shows fallback if no list id 1`] = `
<div>
  Fallback
</div>
`;

exports[`DisplayA component shows message 1`] = `
<span>
  The Words
</span>
`;

exports[`DisplayA component shows message in bold 1`] = `
<span>
  <strong>
    The Words
  </strong>
</span>
`;
